tests/integration/test_integration.py

```diff
@@ -72,102 +72,100 @@ async def initialize_vault(vault: Vault, timeout: int = 60) -> Tuple[str, str]:
         raise TimeoutError("Timed out waiting for Vault to be ready.")
 
     @staticmethod
-    async def deploy_charm(ops_test, charm: Path) -> None:
+    async def deploy_charm(ops_test: OpsTest, charm: Path) -> None:
         """Deploys charm.
 
         Args:
-            ops_test:
+            ops_test: Ops test Framework.
             charm: Charm path.
-
-        Returns:
-            None
         """
         resources = {
             "vault-image": METADATA["resources"]["vault-image"]["upstream-source"],
         }
-        await ops_test.model.deploy(
+        await ops_test.model.deploy(  # type: ignore[union-attr]
             charm,
             resources=resources,
             application_name=APPLICATION_NAME,
             trust=True,
             series="focal",
         )
 
-    async def post_deployment_tasks(self, namespace: str) -> str:
+    @pytest.mark.abort_on_fail
+    @pytest.fixture(scope="module")
+    async def build_and_deploy(self, ops_test: OpsTest):
+        """Builds and deploys vault-k8s charm.
+
+        Args:
+            ops_test: Ops test Framework.
+        """
+        ops_test.destructive_mode = False
+        charm = await ops_test.build_charm(".")
+        await self.deploy_charm(ops_test, charm)
+
+    @pytest.mark.abort_on_fail
```

Comment:
I don't think you need this anymore. Test using this method is decorated with `abort_on_fail` anyway.

Reply:
Why we don't? If that fails why would we want to continue? And sorry for the huge amount of typos, I went to fast.

Reply:
`test_given_no_config_when_post_deployment_tasks_and_authorise_charm_then_status_is_active` uses this method. And it's gonna abort the run if it fails. Can the test succeed if post install fails?

Reply:
Yes, fixed.